world/worlddb.cpp

```diff
@@ -234,40 +234,35 @@ void WorldDatabase::GetCharSelectInfo(uint32 account_id, EQApplicationPacket **o
 			for (auto row_d = results_bind.begin(); row_d != results_bind.end(); ++row_d) {
 				/* If a bind_id is specified, make them start there */
 				if (atoi(row_d[1]) != 0) {
-					player_profile_struct.binds[4].zone_id = (uint32) atoi(row_d[1]);
-					content_db.GetSafePoints(
-						ZoneName(player_profile_struct.binds[4].zone_id, true),
-						0,
-						&player_profile_struct.binds[4].x,
-						&player_profile_struct.binds[4].y,
-						&player_profile_struct.binds[4].z,
-						&player_profile_struct.binds[4].heading
-					);
+					pp.binds[4].zone_id = (uint32) atoi(row_d[1]);
+
+					auto z = GetZone(pp.binds[4].zone_id);
+					if (z) {
+						pp.binds[4].x       = z->safe_x;
+						pp.binds[4].y       = z->safe_y;
+						pp.binds[4].z       = z->safe_z;
+						pp.binds[4].heading = z->safe_heading;
+					}
 				}
 					/* Otherwise, use the zone and coordinates given */
 				else {
-					player_profile_struct.binds[4].zone_id = (uint32) atoi(row_d[0]);
+					pp.binds[4].zone_id = (uint32) atoi(row_d[0]);
 					float x = atof(row_d[2]);
 					float y = atof(row_d[3]);
 					float z = atof(row_d[4]);
 					float heading = atof(row_d[5]);
 					if (x == 0 && y == 0 && z == 0 && heading == 0) {
-						content_db.GetSafePoints(
-							ZoneName(player_profile_struct.binds[4].zone_id, true),
-							0,
-							&x,
-							&y,
-							&z,
-							&heading
-						);
+						auto zone = GetZone(pp.binds[4].zone_id);
+						if (zone) {
+							pp.binds[4].x       = zone->safe_x;
+							pp.binds[4].y       = zone->safe_y;
+							pp.binds[4].z       = zone->safe_z;
+							pp.binds[4].heading = zone->safe_heading;
```

Comment:
There's a logic change here I think might be unintended. The previous code only set `binds[4]` to the safe coordinates if x,y,z,h were all 0, otherwise it set it to the x,y,z,h values. The new code only ever sets it to the safe coordinates

Reply:
Roger that, good eye, will change when I’m coding next. I’ll take another pass at touch points as well

Reply:
Fixed